framework/topology.cpp

```diff
@@ -754,18 +434,30 @@ template <auto &fnArray> static bool try_detection(struct cpu_info *cpu)
     return false;
 }
 
+template <typename CallDetector> static bool call_detector(CallDetector fn, struct cpu_info *cpu, hwloc_topology_t topology = nullptr) {
+    // Check if the fn matches the regular signature
+    if constexpr (std::is_invocable_v<CallDetector, struct cpu_info *>) {
+        return fn(cpu);
+    }
+    // If it matches the specialized signature
+    else if constexpr (std::is_invocable_v<CallDetector, struct cpu_info *, hwloc_topology_t>) {
+        return fn(cpu, topology);
+    }
+    return false;
+}
```

Comment:
Move the `call_detector` function above its use.

But better yet, remove the need for it. Just pass `hwloc_topology_t` to all functions; if they don't need it, then fine.